app/main.py

```diff
@@ -1,2 +1,61 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 0.66) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.length = 0
+        self.limit = capacity * load_factor
+        self.hash_table = [[] for _ in range(capacity)]
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length > self.limit:
+            self.resize()
+        index = self._get_index(key)
+        bucket = self.hash_table[index]
+        for i in range(len(bucket)):
+            if bucket[i][0] == key:
+                bucket[i] = (key, hash(key), value)
+                return
```

Comment:
For this loop?
you know that the key is under the 0 index of the bucket, use this index, no need to use for loop here